core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java

```diff
@@ -41,6 +41,29 @@ public static void extractTons(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWa
 //            logger.warn("Value " + value + " for " + valueEncoder.getName() + " was too large and truncated to " + valueEncoder.getDecimal(false, edgeFlags));
     }
 
+    /**
+     * This parses the weight for a conditional value like "delivery @ (weight > 7.5)"
+     */
+    public static double conditionalWeightToTons(String value) {
+        try {
+            int index = value.indexOf("weight>"); // maxweight or weight
+            if (index < 0) {
+                index = value.indexOf("weight >");
+                if (index > 0) index += "weight >".length();
+            } else {
+                index += "weight>".length();
+            }
+            if (index > 0) {
+                int lastIndex = value.indexOf(')', index); // (value) or value
```

Comment:
```suggestion
                int lastIndex = value.lastIndexOf(')', index); // (value) or value
```

I'm not sure if more complex statements with multiple parentheses are allowed, but using the last parenthesis is correct either way.